src/Akka.Cluster.Hosting.Tests/ClusterSingletonSpecs.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Event;
+using Akka.Hosting;
+using Akka.Remote.Hosting;
+using Akka.TestKit.Xunit2.Internals;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Cluster.Hosting.Tests;
+
+public class ClusterSingletonSpecs
+{
+    public ClusterSingletonSpecs(ITestOutputHelper output)
+    {
+        Output = output;
+    }
+
+    public ITestOutputHelper Output { get; }
+    
+    private class MySingletonActor : ReceiveActor
+    {
+        public static Props MyProps => Props.Create(() => new MySingletonActor());
+
+        public MySingletonActor()
+        {
+            ReceiveAny(_ => Sender.Tell(_));
+        }
+    }
+
+    private async Task<IHost> CreateHost(Action<AkkaConfigurationBuilder> specBuilder, ClusterOptions options)
```

Comment:
Created a custom method for spinning up clustered tests hosts - probably need to move this to its own utility class at some point.